src/MSBuild.Conversion.SDK/SemanticVersion.cs

```diff
@@ -0,0 +1,221 @@
+            {
+                int minor;
+                int.TryParse(versionParts[1], out minor);
+                ver.Minor = minor;
+            }
+
+            if (versionParts.Length > 2)
+            {
+                int patch;
+                int.TryParse(versionParts[2], out patch);
+                ver.Patch = patch;
+            }
+
+            // setup the componentized versions, for later comparison. Nulls are ok, SemanticVersionComponentizedString handles them.
+            ver.BuildMetadataComponents = new SemanticVersionComponentizedString(ver.BuildMetadata);
+            ver.PrereleaseVersionComponents = new SemanticVersionComponentizedString(ver.PrereleaseVersion);
+
+            return ver;
+        }
+
+        public int CompareTo(SemanticVersion other)
+        {
+            if (other == null)
+            {
+                return 1;
+            }
+
+            int result = Major.CompareTo(other.Major);
+
+            if (result != 0)
+            {
+                return result;
+            }
+
+            result = Minor.CompareTo(other.Minor);
+
+            if (result != 0)
+            {
+                return result;
+            }
+
+            result = Patch.CompareTo(other.Patch);
+
+            if (result != 0)
+            {
+                return result;
+            }
+
+            if (PrereleaseVersionComponents.Parts.Count == 0 && other.PrereleaseVersionComponents.Parts.Count > 0)
+            {
+                return 1;
+            }
+            else if (PrereleaseVersionComponents.Parts.Count > 0 && other.PrereleaseVersionComponents.Parts.Count == 0)
+            {
+                return -1;
+            }
+            // only compare the prerel parts if they both have them (or neither, which will compare equal
+            result = PrereleaseVersionComponents.CompareTo(other.PrereleaseVersionComponents);
+            if (result != 0)
+            {
+                return result;
+            }
+
+            if (BuildMetadataComponents.Parts.Count == 0 && other.BuildMetadataComponents.Parts.Count > 0)
+            {
+                return 1;
+            }
+            else if (BuildMetadataComponents.Parts.Count > 0 && other.BuildMetadataComponents.Parts.Count == 0)
+            {
+                return -1;
+            }
+            result = BuildMetadataComponents.CompareTo(other.BuildMetadataComponents);
+            if (result != 0)
+            {
+                return result;
+            }
+
+            string thisOriginalText = OriginalText?.TrimStart(LeadingVersionChars) ?? string.Empty;
+            string otherOriginalText = other.OriginalText?.TrimStart(LeadingVersionChars) ?? string.Empty;
+
+            return StringComparer.OrdinalIgnoreCase.Compare(thisOriginalText, otherOriginalText);
+        }
+
+        public bool Equals(SemanticVersion other) => other != null && string.Equals(OriginalText, other.OriginalText, StringComparison.OrdinalIgnoreCase);
```

Comment:
```suggestion
        public bool Equals(SemanticVersion other) => other is { } && string.Equals(OriginalText, other.OriginalText, StringComparison.OrdinalIgnoreCase);
```

we use `is { }` in this house